### PR TITLE
fix(headless): ensure each result keep a reference of its 'source' searchId

### PR DESCRIPTION
**Pull request by y-lakhdar**
Make sure to keep the id of the search that returned each result.

https://coveord.atlassian.net/browse/KIT-3481

### CODE REVIEW

File: packages/headless/src/api/search/search/result.ts [hunk 1]
```diff
@@ -177,4 +177,9 @@ export interface Result {
    * Whether the result item has been previously viewed by one of the users specified in the `canSeeUserProfileOf` section of the [search token](https://docs.coveo.com/en/13/api-reference/search-api#tag/Search-V2/operation/token) generated to perform the search request.
    */
   isUserActionView: boolean;
+
+  /**
+   * Represents the unique identifier of the search that returned this result.
```
Comment (fbeaudoincoveo) [inline packages/headless/src/api/search/search/result.ts:182]:
```suggestion
   * The unique identifier of the search that returned this result.
```

File: packages/headless/src/features/instant-results/instant-result-analytics-actions.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@
       const docInfo = partialDocumentInformation(result, state);
       const docId = documentIdentifier(result);
       return {
-        searchUid: state.search?.response.searchUid ?? '',
+        searchUid: result.searchUid ?? '',
```
[the remark this answers is not in the document]
Comment (fbeaudoincoveo) [reply packages/headless/src/features/instant-results/instant-result-analytics-actions.ts:39]: psh

**Review by fbeaudoincoveo — APPROVED** [review]
LGTM but we also need to do it in the commerce sub-package 😅

**Review by louis-bompart — APPROVED** [review]
👍 Can you backport this on v2

**Pull request merged**